flow/project.py

```diff
@@ -1265,18 +1254,18 @@ def _get_run_ops(ignore_ops, additional_ignores_flag):
                 Runnable operations.
 
             """
-            return list(
-                set(
-                    self._create_run_job_operations(
-                        entrypoint=entrypoint,
-                        default_directives=default_directives,
-                        jobs=jobs,
-                        ignore_conditions=ignore_conditions_on_execution
-                        | additional_ignores_flag,
-                    )
+            ignore_ops = set(ignore_ops)
+            return [
+                op
+                for op in self._create_run_job_operations(
+                    entrypoint=entrypoint,
+                    default_directives=default_directives,
+                    jobs=jobs,
+                    ignore_conditions=ignore_conditions_on_execution
+                    | additional_ignores_flag,
                 )
-                - set(ignore_ops)
-            )
+                if op not in ignore_ops
+            ]
```

Comment:
This is a crucial change - the previous `set(...) - set(...)` approach does not order the results deterministically, because the `set` class in Python is unordered. This led to inconsistency between runs with template tests. We must use a `list` to iterate over the created run-job-operations, but it's perfectly fine to use a `set` for the ignored operations because it's O(1) to check `if op not in ignore_ops` if `ignore_ops` is a `set`.

Reply:
The problem was nondeterminism in this block of code, which the tests showed. We ended up printing out the job operations in the section `# Eligible to run:` and using a difference of (unordered) sets made that output have no guaranteed order. Switching to a list comprehension that excludes items in the `ignore_ops` set makes the order consistent from run to run. We need ordering for test stability and workflow reproducibility.